### ISSUE TITLE
Realm.getInstance(RealmConfiguration) error

### ISSUE THREAD

**Issue by lourenci** [realm/realm-java#2416]
Hi, I've updated Realm from 0.87.5 to 0.88.00 and I am getting a NullPointerException error when I try to open a new instance without close the previous opened instance.

I have an Activity that I open a Realm instance on onCreate and I close it on onDestroy, but I have to start an IntentService and when this IntentService open an another Realm instance, I get a NullPointerException. If I close the realm of Activity before onDestroy, nothing happens.

```
        config = new RealmConfiguration.Builder(context)
                .name(Environment.REALM_NAME)
                .schemaVersion(Environment.REALM_VERSION)
                .build();

        realm = Realm.getInstance(config);
```

I am getting a NullPointerException when I open Realm.getInstance(config) twice without close the first one. If I close the first instance, it is ok.

**Comment by zaki50** [comment]
@lourenci thank you for your report. I'll look into this.

**Comment by zaki50** [comment]
It seems to happen when RxJava is not available. I confirmed it's a bug of Realm.

**Comment by zaki50** [comment]
Work-around:

Use **single** `RealmConfiguration` instance when calling `Realm.getInstance(config);`

Or use `Realm.setDefaultConfiguration()` and `Realm.getDefaultInstance()`

Those are recommended way of `RealmConfiguration` usage.

**Comment by zaki50** [comment]
I merged #2419 into `releases` branch and #2422 into `master` branch.

I'll close this issue.
If you have further question, please reopen this @lourenci

**Comment by lourenci** [comment]
Tks 😀
